base/MAPL_IO.F90

```diff
@@ -2935,28 +2939,34 @@ subroutine MAPL_FieldWriteNCPar(formatter, name, FIELD, ARRDES, HomePE, oClients
              endif
 
              if (arrdes%write_restart_by_oserver) then
-                if(arrdes%writers_comm /= MPI_COMM_NULL) then
+                if(MAPL_AM_I_ROOT()) then
```

Comment:
These `MAPL_AM_I_ROOT()` references seem very unusual.  These are more typically for allocating buffers on the root process and/or log messages.

I'll approve, but would appreciate an explanation.

Reply:
By default, ArrayGather gathers everything to root. But I found it has optional argument "depe" ( destination PE , I guess). I am wondering to make changes to link the writing PE with the "pede"

Reply:
The difficulty here is the client does not ( or should not)  know about oserver